### PR TITLE
rec: add interface (not subject to proxy protocol substitutions) addresses in Lua DNSQuestion and corresponding FFI

### LINKED ISSUE #13730: REC: When using PROXY, provide the "real" localaddr somewhere
- Program: Recursor
 - Issue type: Feature request

### Short description
When using the Proxy protocol, `dq.localaddr` and `dq.remoteaddr` contain the address the client connected to, and the client's ip.

While this makes ACLs etc real easy, sometimes we need the "local" localaddr and "local" remoteaddr.

### Usecase
Imagine a setup with one or more dnsdist proxies and a number of pdns-rec backends, using the PROXY protocol. Each backend-server could have multiple backend IP addresses.

For debugging/logging/metric-gathering purposes, we'd need to log not only the client's ip address (`dq.remoteaddr`), the ip the client connected to (`dq.localaddr`), but also the local ip on the backend which processed the query. 

### Description
Note that as the backend might have multiple IPs accepting queries, just the "primary" ip or hostname might not be enough in the logs.

Suggestion would be to add e.g. `real_localaddr` and `real_remoteaddr` to the `dq` object when removing the PROXY headers. That removes a bit of the PROXY transparancy, but I feel this would be justified in this case.

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Fixes #13730

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [X] included documentation (including possible behaviour changes)
- [ ] documented the code
- [X] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Review by rgacogne — COMMENTED** [review]
Code looks good but I'm not fond of `physical` which is a bit confusing to me, as I would expect an ethernet address. DNSdist uses `hopRemote` and `hopLocal` internally which is not great either, to be honest.

**Comment by omoerbeek** [comment]
Te name is confusing indeed. Pondering original/real/interface...

**Comment by omoerbeek** [comment]
I'm leaning towards `interface_remoteaddr` and `interface_localaddr`. @rgacogne  @Habbie how about that?

**Comment by Habbie** [comment]
sounds good to me

**Comment by franklouwers** [comment]
Keeping Physical would be consistent with https://doc.powerdns.com/recursor/lua-config/proxymapping.html (note that I don't like the naming there either, if we change it here, we might consider changing it there as well to remain consistent)

**Comment by omoerbeek** [comment]
> Keeping Physical would be consistent with https://doc.powerdns.com/recursor/lua-config/proxymapping.html (note that I don't like the naming there either, if we change it here, we might consider changing it there as well to remain consistent)

The term "physical" is only used in the docs there, it is not something that appears in the current Lua scripting/config API. I'll add a commit to change the terminology there as well..

File: pdns/recursordist/docs/lua-scripting/dq.rst [hunk 1]
```diff
@@ -50,10 +50,20 @@ The DNSQuestion object contains at least the following fields:
...
+      If the proxy protocol is used, this will contain the source address from the proxy protocol header.
 
   .. attribute:: DNSQuestion.localaddr
 
       :class:`ComboAddress` where this query was received on.
+      If the proxy protocol is used, this will contain the destination address from the proxy protocol header.
+
+  .. attribute:: DNSQuestion.interface_remoteaddr
+
+      :class:`ComboAddress` of the interface the requestor used, that is, the network source address of the request.
```
Comment (rgacogne) [inline pdns/recursordist/docs/lua-scripting/dq.rst:62]:
```suggestion
      Source :class:`ComboAddress` of the packet received by the recursor. If the proxy protocol is not used, the value will match ``remoteaddr``.
```

File: pdns/recursordist/docs/lua-scripting/dq.rst [hunk 2]
```diff
@@ -50,10 +50,20 @@ The DNSQuestion object contains at least the following fields:
...
+
+  .. attribute:: DNSQuestion.interface_localaddr
+
+      The :class:`ComboAddress` of the interface where this query was received on, which is one of the listening addresses of the recursor.
```
Comment (rgacogne) [inline pdns/recursordist/docs/lua-scripting/dq.rst:66]:
```suggestion
      Destination :class:`ComboAddress` of the packet received by the recursor. If the proxy protocol is not used, the value will match ``localaddr``.
```

**Review by rgacogne — APPROVED** [review]
Looks good, one suggestion for the documentation.

**Pull request merged**